proto/one/enterprise/data/dataset_definition_v2.proto

```diff
@@ -0,0 +1,72 @@
+syntax = "proto3";
+
+import "google/protobuf/wrappers.proto";
+
+option csharp_namespace ="ONE.Models.CSharp.Enterprise.Data";
+
+message EnterpriseDataSourceDefinition {
+    // DataSoruce name
+    string name = 1;
+
+    // Optional: A list of key value configuration options for the datasource data retrieval
+    map<string, google.protobuf.StringValue> options = 2;
```

Comment:
should this be <string, string>?

Reply:
Yea, unless you plan to have different behavior if the string is null vs empty I would just use <string, string>.

Reply:
It can. 
When I quick learn StringValue vs string, the difference is the StirngValue can be null, and string in protobuf cannot be null, I guess, we can enforce dataSource option arguments only pass with values.

Reply:
If you use string the default value is an empty string, if you use StringValue the default value is null. Unless you need to have different behavior when someone sets the string to empty vs not providing a value, there is no real need to use stringValue. If your logic is if(string.IsNullOrEmpty("value")) then it will be true on both string and StringValue defaults.